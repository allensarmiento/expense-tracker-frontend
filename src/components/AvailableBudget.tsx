import styled from 'styled-components';
import BannerSection from './BannerSection';
import BannerTitle from './BannerTitle';
import AmountDisplay from './AmountDisplay';

const Total = styled.div`
  font-size: 3.5rem;
  text-align: center;
`;

const Detail = styled.div`
  width: 100%;
`;

type Totals = {
  available: number;
  income: number;
  expense: number;
};

type AvailableBudgetProps = {
  totals: Totals;
};

const AvailableBudget = ({ totals }: AvailableBudgetProps) => {
  return (
    <BannerSection justifyContent="space-between">
      <BannerTitle>Total Available Budget</BannerTitle>
      <Total>{totals.available}</Total>
      <Detail>
        <AmountDisplay amount={totals.income} color="green" />
        <AmountDisplay amount={totals.expense} color="red" />
      </Detail>
    </BannerSection>
  );
};

export default AvailableBudget;
