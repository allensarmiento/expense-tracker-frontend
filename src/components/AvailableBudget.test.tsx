import { shallow } from 'enzyme';
import AvailableBudget from './AvailableBudget';

it('renders without crashing', () => {
  expect(shallow(
    <AvailableBudget totals={{
      available: '10.00',
      income: '15.00',
      expense: '5.00'
    }} />
  )).toMatchSnapshot();
});
