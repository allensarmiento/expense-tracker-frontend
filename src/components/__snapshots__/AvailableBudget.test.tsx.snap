// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BannerSection
  justifyContent="space-between"
>
  <BannerTitle>
    Total Available Budget
  </BannerTitle>
  <styled.div>
    $
    10.00
  </styled.div>
  <styled.div>
    <AmountDisplay
      amount="15.00"
      color="green"
    />
    <AmountDisplay
      amount="5.00"
      color="red"
    />
  </styled.div>
</BannerSection>
`;
